### ISSUE TITLE
Popover `onOpen` called on each re-render.

### ISSUE THREAD

**Issue by crabvk** [mantinedev/mantine#7111]
### Dependencies check up

- [X] I have verified that I use latest version of all @mantine/* packages

### What version of @mantine/* packages do you have in package.json?

7.13.5

### What package has an issue?

@mantine/core

### What framework do you use?

Vite

### In which browsers you can reproduce the issue?

All

### Describe the bug

Update component state with `Popover` opened, `onOpen` triggered on each re-render.
Expected `onOpen` to be called only once when I open `Popover` (as it works now) and nothing on re-renders (the bug).

### If possible, include a link to a codesandbox with a minimal reproduction

https://codesandbox.io/p/sandbox/mantine-popover-bug-forked-8446wc

### Self-service

- [ ] I would be willing to implement a fix for this issue

**Comment by rtivital** [comment]
Your sandbox is private, I cannot access it, please make it public.

**Comment by maxiadlovskii** [comment]
I created another issue #7115  I think they might have the same root

**Comment by rtivital** [comment]
Fixed in [7.14.1](https://github.com/mantinedev/mantine/releases/tag/7.14.1)
